src/components/hooks/useElementPosition.js

```diff
@@ -0,0 +1,31 @@
+import {useEffect, useState} from 'react';
+
+/**
+ * Custom hook that receives an element reference and returns the position of the element
+ *
+ * @param {object} elementRef  Reference to the element
+ *
+ * @returns {DOMRect} Element position
+ */
+export default function useElementPosition(elementRef) {
+  const [position, setPosition] = useState(null);
+  const target = elementRef.current;
+
+  if (target && !position) {
+    setPosition(target.getBoundingClientRect());
+  }
+
+  useEffect(() => {
+    const onWindowResize = () => {
+      if (target) {
+        setPosition(target.getBoundingClientRect());
+      }
+    };
+
+    window.addEventListener('resize', onWindowResize);
```

Comment:
there is no point in adding a resize handler if we have no target, we'd better wait for the target to show up:
```js
let cleanup;

if (target) {
  const onwindowresize = ... (without if);

  window.add;
  cleanup = window.remote;
}

return cleanup;